test/parallel/test-worker-message-port-transfer-self.js

```diff
@@ -0,0 +1,33 @@
+// Flags: --experimental-worker
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const { MessageChannel } = require('worker_threads');
+
+const { port1, port2 } = new MessageChannel();
+
+assert.throws(common.mustCall(() => {
+  port1.postMessage(null, [port1]);
+}), common.mustCall((err) => {
```

Comment:
Can this be an `expectsError`?

Reply:
Per spec, `message` is an enumerable getter on the `DOMException` prototype, rather than a non-enumerable own property on the `DOMException` object, which `expectsError` expects:

https://github.com/nodejs/node/blob/a9d9d7689d326d363c8d9546a80ba76f4cd0decb/test/common/index.js#L704-L706